src/hu_stuff.c

```diff
@@ -1064,7 +1066,24 @@ static void HU_UpdateCrosshair(void)
     if (hud_crosshair_target && crosshair_target
         && !(crosshair_target->flags & MF_SHADOW))
     {
-      crosshair.cr = colrngs[hud_crosshair_target_color];
+      // [Alaux] Color crosshair by target health
+      if (hud_crosshair_target == crosstarget_health)
```

Comment:
First of all, coloring by target health takes precedence over target highlight, since it's basically an enhanced version of the latter - as I mentioned in the Doomworld thread.

You may figure out that coloring by player health and by target health can be used simultaneously. As implied by what's said above, the latter overrides the former, just like what was done with the original target highlight. While I'm pretty sure that this would be confusing in case the same color is applied by both features, they don't necessarily have to be used at the same time, so it's up to the user to decide what behavior they want for their crosshair.